src/main/java/seedu/address/logic/parser/MarkingCommandParser.java

```diff
@@ -0,0 +1,47 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.MarkCommand;
+import seedu.address.logic.commands.MarkingCommand;
+import seedu.address.logic.commands.UnmarkCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new MarkingCommand object
+ */
+public class MarkingCommandParser implements Parser<MarkingCommand> {
+
+    private final String typeOfMarking;
+
+    public MarkingCommandParser(String typeOfMarking) {
+        this.typeOfMarking = typeOfMarking;
+    }
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the DoneCommand
+     * and returns a DoneCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public MarkingCommand parse(String args) throws ParseException {
+        try {
+            Index[] indexes = ParserUtil.parseMultipleIndex(args);
+            if (typeOfMarking.equals("mark")) {
+                return new MarkCommand(indexes);
+            } else {
+                return new UnmarkCommand(indexes);
+            }
+        } catch (ParseException pe) {
+            String errorMessage;
+            if (typeOfMarking.equals("mark")) {
+                errorMessage = MarkCommand.MESSAGE_USAGE;
+            } else {
+                errorMessage = UnmarkCommand.MESSAGE_USAGE;
+            }
```

Comment:
Maybe you could make use of the existing if-else block in the try block? This can be done by declaring the String `errorMessage` outside of the try-catch block:

```java
String errorMessage = null;
try {
    // if else block setting of errorMessage
catch (ParseException pe) {
    // throw new ParseException with errorMessage
}
```

An alternative solution could also be doing this inside the catch block:
`String errorMessage = typeOfMarking.equals(MarkCommand.COMMAND_WORD) ? MarkCommand.MESSAGE_USAGE : UnmarkCommand.MESSAGE_USAGE;`

These are just a few possible suggestions! I'm not sure which are best, you could even stick with your current solution if you think its best!